src/network/MissionControlTasks.cpp

```diff
@@ -0,0 +1,186 @@
+#include "MissionControlTasks.h"
+
+#include "../Constants.h"
+#include "../Globals.h"
+#include "../utils/core.h"
+#include "../world_interface/world_interface.h"
+#include "MissionControlMessages.h"
+
+#include <loguru.hpp>
+
+#include <nlohmann/json.hpp>
+
+using namespace robot::types;
+using namespace std::chrono_literals;
+
+using nlohmann::json;
+
+namespace net::mc::tasks {
+namespace {
+const std::chrono::milliseconds TELEM_REPORT_PERIOD = 100ms;
+}
+
+PowerRepeatTask::PowerRepeatTask()
+	: util::PeriodicTask<>(Constants::JOINT_POWER_REPEAT_PERIOD,
+						   std::bind(&PowerRepeatTask::periodicTask, this)) {}
+
+void PowerRepeatTask::setJointPower(jointid_t id, double power) {
+	std::lock_guard lock(_mutex);
+	_last_joint_power[id] = power;
+}
+
+void PowerRepeatTask::setCmdVel(double steerVel, double xVel) {
+	std::lock_guard lock(_mutex);
+	_last_cmd_vel = {steerVel, xVel};
+}
+
+void PowerRepeatTask::start() {
+	// lock the mutex before start() to prevent data races between start() and stop()
+	std::lock_guard lock(_mutex);
```

Comment:
Does the mutex in the AsyncTask/PeriodicTask class not prevent this?

Reply:
That mutex only protects the members of the super class. Since this class adds its own members that need to be synchronized (since multiple threads could be accessing `_last_cmd_vel` and `_last_joint_power` at the same time, we need to add a mutex in this class to synchronize those accesses.

So, since we lock that mutex and modify the fields in `stop()`, we should correspondingly lock the mutex in `start()`, which ensures that `start()` can't be called while `stop()` is in the middle of executing. Otherwise, that could cause some invalid states.